### PR TITLE
Server and Worker tests

### PR DESCRIPTION
**Pull request by PClmnt**
## Description
Added multiple new server and worker tests.

Also have some other controller tests in the server to come but will get them up in a seaprate PR as I ended up tweaking them with the changes I made to how the server runs the tests. 

The main gaps now are external datasource tests. i.e `src/api/controllers/row` and  `src/integrations` but hopefully with the addition of the test image these wil become a lot easier to write tests for.

### CODE REVIEW

File: packages/server/src/api/routes/tests/cloud.spec.ts [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  })
+
+  describe("import", () => {
+    it("should be able to import apps", async () => {
+      const res = await request
+        .post(`/api/cloud/import`)
+        .attach("importFile", "src/api/routes/tests/data/export-test.tar.gz")
+        .set(config.defaultHeaders())
+        .expect(200)
+      expect(res.body.message).toEqual("Apps successfully imported.")
```
Comment (mike12345567) [inline packages/server/src/api/routes/tests/cloud.spec.ts:37]: In this test, would it be possible to query to see if any of the apps had been created?

File: packages/worker/src/api/routes/global/tests/license.spec.ts [hunk 2]
```diff
@@ -18,7 +20,10 @@ describe("/api/global/license", () => {
   })
 
   describe("POST /api/global/license/activate", () => {
-    it("activates license", () => {})
+    it("activates license", async () => {
+      //let result = await config.api.license.activate("test")
```
Comment (mike12345567) [inline packages/worker/src/api/routes/global/tests/license.spec.ts:24]: Again just wondering why commented out

File: packages/worker/src/api/routes/global/tests/templates.spec.ts [hunk 3]
```diff
@@ -1,4 +1,17 @@
...
+  EmailTemplatePurpose,
+  TemplateMetadata,
+  TemplateMetadataNames,
+  TemplateType,
+} from "../../../../constants"
 import { TestConfiguration } from "../../../../tests"
+import { join } from "path"
+import { readStaticFile } from "../../../../../src/utilities/fileSystem"
 
 // TODO
```
Comment (mike12345567) [inline packages/worker/src/api/routes/global/tests/templates.spec.ts:16]: can probably remove the `TODO` as its done now!

File: packages/worker/src/constants/index.ts [hunk 4]
```diff
@@ -27,6 +27,14 @@ export enum EmailTemplatePurpose {
   CUSTOM = "custom",
 }
 
+export enum TemplateMetadataNames {
```
Comment (mike12345567) [inline packages/worker/src/constants/index.ts:30]: Enums should have singular names `TemplateMetadataName`

**Review by mike12345567 — APPROVED** [review]
Great set of tests, amazing work on this Peter! Just a couple of comments about tests which have been commented out, as well as the import test case, would be interesting to know if we could check if anything had been imported or not!

**Review by Rory-Powell — COMMENTED** [review]
Nice work peter! Just one comment about the `pro/tests`.

File: packages/worker/jest.config.ts [hunk 5]
```diff
@@ -19,6 +19,8 @@ if (!process.env.CI) {
   }
   // add pro sources if they exist
   if (fs.existsSync("../../../budibase-pro")) {
+    config.moduleNameMapper["@budibase/pro/(.*)"] =
+      "<rootDir>/../../../budibase-pro/packages/pro/$1"
```
[the remark this answers is not in the document]
Comment (Rory-Powell) [reply packages/worker/jest.config.ts:23]: Actually nvm, just noticing the check above 👍  

That should mean open source people would use the built versions of `@budibase/pro/tests` they pull from npm. I'm not sure we will include that in the build by default, we will have to update:

https://github.com/Budibase/budibase-pro/blob/develop/packages/pro/tsconfig.build.json#L14

To be:
```json
  "include": ["./src/**/*", "./tests/**/*"],
```

**Pull request merged**